index.js

```diff
@@ -56,42 +59,19 @@ Client.encoding = Object.freeze({
 function Client (opts) {
   if (!(this instanceof Client)) return new Client(opts)
 
-  this.config(opts)
-
-  Writable.call(this, this._conf)
-
-  const errorproxy = (err) => {
-    if (this.destroyed === false) this.emit('request-error', err)
-  }
-
-  const fail = () => {
-    if (this._writableState.ending === false) this.destroy()
-  }
+  Writable.call(this, { objectMode: true })
```

Comment:
Reviewer note: The only "Writable" option we every pass in `objectMode`, so let's only pass that. And it doesn't need to be (and shouldn't be) externally configurable.

Reply:
Just to say it out loud, this change means that _if_ someone was using the `apm-nodejs-http-client` package stand-alone to talk to APM Server that they no longer have the ability to control which configuration/opts values the underlying writable stream is instantiated with.  I don't think this is a blocker -- it's an esoteric use-case on top of an esoteric use-case and if someone runs into it they can easily rollback to an older version and open an issue here.  I also don't think we've promised that the configuration interface to this package is covered under semver so I don't think we need a breaking change bump.  Finally, it also seems like _the right_ thing to do -- locking down which configuration values the underlying stream uses should make the client/agent behave in am ore stable fashion.

Reply:
If it helps later reviewing, the available Writable options are: https://nodejs.org/api/stream.html#stream_new_stream_writable_options

Thanks for bringing this up. The same potential "issue" exists for the options passed to the underlying HTTP Agent: https://nodejs.org/api/all.html#http_new_agent_options

I'll add a small note about these in the CHANGELOG.md -- which I need to update to sync with master anyway.